source/connection-monitoring-and-pooling/tests/README.rst

```diff
@@ -36,7 +36,21 @@ Unit Test Format:
 
 All Unit Tests have some of the following fields:
 
-- ``poolOptions``: if present, connection pool options to use when creating a pool
+- ``poolOptions``: If present, connection pool options to use when creating a pool;
+  both `standard ConnectionPoolOptions <https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#connection-pool-options-1>`__
+  and the following test-specific options are allowed:
+
+  - ``backgroundThreadDelayMS``: An artificial delay before starting observable activities in a
+    `Background Thread <https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#background-thread>`__.
+    The Test Runner / Connection Pool MUST try to start counting time towards this delay
+    at an instant as close to starting executing ``operations`` as reasonably possible.
+    If a Connection Pool does not implement a Background Thread, the Test Runner MUST ignore the option.
+    Possible values:
+
+    - 0 (default)—no delay;
```

Comment:
After looking at the changes, the only value that seems to be needed is -1, so I'm wondering if it would be better to make this just a boolean indicating whether the background thread should run at all. I'm struggling to come up with a case where a hard-coded delay would actually be used.

Reply:
@patrickfreed maybe this boolean flag can be replaced by [frequency setting](https://jira.mongodb.org/browse/DRIVERS-1465) with 0 or -1 values?

Reply:
Yeah that sounds like a good idea. That'll also allow us to remove the 50ms guidance from the README and move it to the tests that require it.

Reply:
I agree that we don't need to implement DRIVERS-1465 now. Evolving switch option to frequency option in DRIVERS-1465 sounds good to me. @patrickfreed I believe there is no concern of significant overhead in evolving switch to frequency option in driver implementations?

Reply:
I agree that we don't need to implement DRIVERS-1465 now, but I do think we should implement the frequency option now (for tests only) to prevent churn in the various drivers' test runners in the future. For an individual driver it isn't _significant_ overhead to make this change and then later convert to the frequency option, but when multiplied across every language team it is significant for the DBX team as a whole. Also, if a driver lags in converting its runner to the frequency option, it could end up in a state where it can't sync new unrelated tests due to an outdated runner. Also, existing runners likely have support for tuning the frequency anyways, as the README currently requires it be set to 50ms. The proposed option would just make that variable per test, whereas the on/off flag could require different codepaths.

Reply:
The concept of frequency as it is currently mentioned in the test runner spec is vague. I introduced `backgroundThreadRunPeriodMS`, which hopefully meets the desired intent.